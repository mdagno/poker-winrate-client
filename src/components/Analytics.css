.analysis select {
  width: 40%;
  margin-left: auto;
  margin-right: auto;
}

.analysis {
  position: absolute;
  margin: 10px;
  width: 80%;
  height: 400px;
  display: flex;
  flex-direction: column;
  text-align: center;
  margin-left: auto;
  margin-right: auto;
  color: white;
  background-color: white;
}