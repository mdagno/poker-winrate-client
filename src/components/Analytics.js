import React from 'react'
import './Analytics.css'
import { VictoryAxis } from 'victory';
import { VictoryLine } from 'victory';
import { VictoryTheme } from 'victory';
import { VictoryChart } from 'victory';
import ApiContext from '../ApiContext';

export default class Analytics extends React.Component {
state = {
  filter: 'Profits',
}
static contextType = ApiContext;

  generateData = () => {
    const sessionsList = this.context.sessionsList;
    let filter = this.state.filter;
    let renderedData;
    if (filter === 'Profits'){
      renderedData = sessionsList.map((session, index) => {
        return {x: index + 1, y: session.cashed_out - session.buy_in}
      })
    }
    else if(filter === 'Online'){
      let filteredData = sessionsList.filter(session => session.game_type_one === 'Online')
      renderedData = filteredData.map((session, index) => {
        return {x: index + 1, y: session.cashed_out - session.buy_in}
      })
    }
    else if(filter === 'Live'){
      let filteredData = sessionsList.filter(session => session.game_type_one === 'Live')
      renderedData = filteredData.map((session, index) => {
        return {x: index + 1, y: session.cashed_out - session.buy_in}
      })
    }
    console.log()
    
    return renderedData;
  }

  calculateDomain = () => {
    let sessionProfits = this.generateData().map(session => Math.abs(session.y))
    let yDomain = Math.max(...sessionProfits)
    return yDomain;
  }

  handleFilterChange = filterValue => {
    this.setState({
      filter: filterValue,
    })
  }

  render(){
    console.log(this.generateData())
    console.log(this.calculateDomain())
    console.log(this.state)
    return(
      <div className="analysis">
        <label for="filterAnalytics">See Data: </label>
        <select className="filterAnalytics" onChange={e => this.handleFilterChange(e.target.value)}>
          <option value="Profits">Total Profits</option>
          <option value="Online">Online Profits</option>
          <option value="Live">Live Profits</option>
        </select>
        <VictoryChart  height={500} width={450} padding={70} animate={1000} tickValues={[]}>
          <VictoryLine 
            interpolation='linear'
            data={this.generateData()} 
            style={
              {
                data: {
                  stroke: '#09d3ac', 
                  strokeWidth: 1
                },
                parent: {border: "1px solid #ccc"}
            }}
          />
        </VictoryChart>
      </div>
    )
  }
}