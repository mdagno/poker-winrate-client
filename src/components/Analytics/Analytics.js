import React from 'react'
import './Analytics.css'
import { VictoryLine } from 'victory';
import { VictoryChart } from 'victory';
import { VictoryLabel } from 'victory';
import {VictoryTheme} from 'victory';
import ApiContext from '../../ApiContext';

export default class Analytics extends React.Component {
state = {
  filter: 'Profits',
}
static contextType = ApiContext;

  generateData = () => {
    const sessionsList = this.context.sessionsList;
    let filter = this.state.filter;
    let renderedData = [];
    let accTotal = 0;

    if (filter === 'Profits'){
      sessionsList.forEach((session, index) => {
        let sessionProfit = session.cashed_out - session.buy_in;
        accTotal = accTotal + sessionProfit;
        renderedData.push(accTotal);
      })
      return renderedData;
    }

    else if(filter === 'Online Profits'){
      let filteredData = sessionsList.filter(session => session.game_type_one === 'Online')
      filteredData.forEach((session, index) => {
        let sessionProfit = session.cashed_out - session.buy_in;
        accTotal = accTotal + sessionProfit;
        renderedData.push(accTotal);
      })
      return renderedData
    }

    else if(filter === 'Live Profits'){
      let filteredData = sessionsList.filter(session => session.game_type_one === 'Live')
      filteredData.map((session, index) => {
        let sessionProfit = session.cashed_out - session.buy_in;
        accTotal = accTotal + sessionProfit;
        renderedData.push(accTotal);
      })
      return renderedData;
    }

    else if(filter === 'BB/hour'){
      let totalWinRate = 0;
      sessionsList.forEach((session, index) => {
        let sessionWinRate = (session.cashed_out - session.buy_in) / session.big_blind / session.session_length;
        totalWinRate = (totalWinRate + sessionWinRate) / (index + 1);
        renderedData.push(totalWinRate);
      })
      return renderedData;
    }
  }

  handleFilterChange = filterValue => {
    this.setState({
      filter: filterValue,
    })
  }

  render(){
    return(
      <div className="analysis">
        <select className="filterAnalytics" onChange={e => this.handleFilterChange(e.target.value)}>
          <option value="Profits">Total Profits</option>
          <option value="Online Profits">Online Profits</option>
          <option value="Live Profits">Live Profits</option>
          <option value="BB/hour">BB/hour</option>
        </select>
        <VictoryChart  
          height={500} 
          width={450} 
          padding={70}
          animate={
              {
                duration: 1000,
                onLoad: { duration: 1500 }
              }
            } 
          theme={VictoryTheme.material}>
          <VictoryLine 
            interpolation='linear'
            data={this.generateData()} 
            style={
              {
                data: {
                  stroke: '#009BB0', 
                  strokeWidth: 3
                },
                parent: {border: "1px solid #ccc"},
            }}
          />
          <VictoryLabel 
          text='Sessions'
          x={200}
          y={470}
          />
          <VictoryLabel 
          text={this.state.filter}
          x={25}
          y={30}
          />
        </VictoryChart>
      </div>
    )
  }
}