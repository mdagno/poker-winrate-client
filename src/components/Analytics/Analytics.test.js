import React from 'react';
import renderer from 'react-test-renderer';
import Analytics from './Analytics';
import ReactDOM from 'react-dom'


describe('Analytics component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Analytics />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('Analytics renders correctly', () => {
    const tree = renderer
      .create(<Analytics />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
