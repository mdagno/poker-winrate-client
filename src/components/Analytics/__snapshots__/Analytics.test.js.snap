// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics component Analytics renders correctly 1`] = `
<div
  className="analysis"
>
  <select
    className="filterAnalytics"
    onChange={[Function]}
  >
    <option
      value="Profits"
    >
      Total Profits
    </option>
    <option
      value="Online Profits"
    >
      Online Profits
    </option>
    <option
      value="Live Profits"
    >
      Live Profits
    </option>
    <option
      value="BB/hour"
    >
      BB/hour
    </option>
  </select>
  <div
    className="VictoryContainer"
    style={
      Object {
        "height": "100%",
        "pointerEvents": "none",
        "position": "relative",
        "touchAction": "none",
        "userSelect": "none",
        "width": "100%",
      }
    }
  >
    <svg
      height={500}
      role="img"
      style={
        Object {
          "height": "auto",
          "pointerEvents": "all",
          "width": "100%",
        }
      }
      viewBox="0 0 450 500"
      width={450}
    >
      <g
        clipPath="url(#victory-clip-4)"
        style={
          Object {
            "border": "1px solid #ccc",
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <defs>
          <clipPath
            id="victory-clip-4"
          >
            <rect
              height={360}
              vectorEffect="non-scaling-stroke"
              width={0}
              x={70}
              y={70}
            />
          </clipPath>
        </defs>
        <path
          d={null}
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "transparent",
              "opacity": 1,
              "stroke": "#009BB0",
              "strokeWidth": 3,
            }
          }
        />
      </g>
      <text
        direction="inherit"
        dx={0}
        dy={4.97}
        x={200}
        y={470}
      >
        <tspan
          dx={0}
          style={
            Object {
              "fill": "#252525",
              "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
              "fontSize": 14,
              "parent": Object {
                "height": "100%",
                "userSelect": "none",
                "width": "100%",
              },
              "stroke": "transparent",
            }
          }
          textAnchor="start"
          x={200}
        >
          Sessions
        </tspan>
      </text>
      <text
        direction="inherit"
        dx={0}
        dy={4.97}
        x={25}
        y={30}
      >
        <tspan
          dx={0}
          style={
            Object {
              "fill": "#252525",
              "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
              "fontSize": 14,
              "parent": Object {
                "height": "100%",
                "userSelect": "none",
                "width": "100%",
              },
              "stroke": "transparent",
            }
          }
          textAnchor="start"
          x={25}
        >
          Profits
        </tspan>
      </text>
      <g
        role="presentation"
      >
        <line
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "transparent",
              "stroke": "#90A4AE",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeWidth": 2,
            }
          }
          vectorEffect="non-scaling-stroke"
          x1={70}
          x2={380}
          y1={430}
          y2={430}
        />
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={132}
            x2={132}
            y1={430}
            y2={70}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={132}
            x2={132}
            y1={430}
            y2={435}
          />
          <text
            direction="inherit"
            dx={0}
            dy={10.26}
            id="chart-axis-3-tickLabels-0"
            x={132}
            y={443}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={132}
            >
              0.2
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={194}
            x2={194}
            y1={430}
            y2={70}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={194}
            x2={194}
            y1={430}
            y2={435}
          />
          <text
            direction="inherit"
            dx={0}
            dy={10.26}
            id="chart-axis-3-tickLabels-1"
            x={194}
            y={443}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={194}
            >
              0.4
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={256}
            x2={256}
            y1={430}
            y2={70}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={256}
            x2={256}
            y1={430}
            y2={435}
          />
          <text
            direction="inherit"
            dx={0}
            dy={10.26}
            id="chart-axis-3-tickLabels-2"
            x={256}
            y={443}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={256}
            >
              0.6
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={318}
            x2={318}
            y1={430}
            y2={70}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={318}
            x2={318}
            y1={430}
            y2={435}
          />
          <text
            direction="inherit"
            dx={0}
            dy={10.26}
            id="chart-axis-3-tickLabels-3"
            x={318}
            y={443}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={318}
            >
              0.8
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={380}
            x2={380}
            y1={430}
            y2={70}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={380}
            x2={380}
            y1={430}
            y2={435}
          />
          <text
            direction="inherit"
            dx={0}
            dy={10.26}
            id="chart-axis-3-tickLabels-4"
            x={380}
            y={443}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="middle"
              x={380}
            >
              1.0
            </tspan>
          </text>
        </g>
      </g>
      <g
        role="presentation"
      >
        <line
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "transparent",
              "stroke": "#90A4AE",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeWidth": 2,
            }
          }
          vectorEffect="non-scaling-stroke"
          x1={70}
          x2={70}
          y1={70}
          y2={430}
        />
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={380}
            y1={358}
            y2={358}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={65}
            y1={358}
            y2={358}
          />
          <text
            direction="inherit"
            dx={0}
            dy={4.26}
            id="chart-axis-4-tickLabels-0"
            x={57}
            y={358}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="end"
              x={57}
            >
              0.2
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={380}
            y1={286}
            y2={286}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={65}
            y1={286}
            y2={286}
          />
          <text
            direction="inherit"
            dx={0}
            dy={4.26}
            id="chart-axis-4-tickLabels-1"
            x={57}
            y={286}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="end"
              x={57}
            >
              0.4
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={380}
            y1={214}
            y2={214}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={65}
            y1={214}
            y2={214}
          />
          <text
            direction="inherit"
            dx={0}
            dy={4.26}
            id="chart-axis-4-tickLabels-2"
            x={57}
            y={214}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="end"
              x={57}
            >
              0.6
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={380}
            y1={142}
            y2={142}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={65}
            y1={142}
            y2={142}
          />
          <text
            direction="inherit"
            dx={0}
            dy={4.26}
            id="chart-axis-4-tickLabels-3"
            x={57}
            y={142}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="end"
              x={57}
            >
              0.8
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "none",
                "pointerEvents": "painted",
                "stroke": "#ECEFF1",
                "strokeDasharray": "10, 5",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={380}
            y1={70}
            y2={70}
          />
          <line
            role="presentation"
            shapeRendering="auto"
            style={
              Object {
                "fill": "transparent",
                "size": 5,
                "stroke": "#90A4AE",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              }
            }
            vectorEffect="non-scaling-stroke"
            x1={70}
            x2={65}
            y1={70}
            y2={70}
          />
          <text
            direction="inherit"
            dx={0}
            dy={4.26}
            id="chart-axis-4-tickLabels-4"
            x={57}
            y={70}
          >
            <tspan
              dx={0}
              style={
                Object {
                  "fill": "#455A64",
                  "fontFamily": "'Roboto', 'Helvetica Neue', Helvetica, sans-serif",
                  "fontSize": 12,
                  "letterSpacing": "normal",
                  "padding": 8,
                  "stroke": "transparent",
                  "strokeWidth": 0,
                }
              }
              textAnchor="end"
              x={57}
            >
              1.0
            </tspan>
          </text>
        </g>
      </g>
    </svg>
    <div
      style={
        Object {
          "height": "auto",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
          "zIndex": 99,
        }
      }
    >
      <svg
        height={500}
        style={
          Object {
            "height": "auto",
            "overflow": "visible",
            "width": "100%",
          }
        }
        viewBox="0 0 450 500"
        width={450}
      />
    </div>
  </div>
</div>
`;
